core/src/components/picker-column/test/standalone/picker-column.e2e.ts

```diff
@@ -16,5 +18,29 @@ configs({ modes: ['ios'], directions: ['ltr'] }).forEach(({ title, config }) =>
 
       await expect(picker).toBeVisible();
     });
+
+    test('should have the correct selectedIndex and prevSelected', async ({ page }) => {
+      test.info().annotations.push({
+        type: 'issue',
+        description: 'https://github.com/ionic-team/ionic-framework/issues/21764',
+      });
+
+      const ionPickerDidPresent = await page.spyOnEvent('ionPickerDidPresent');
+      const ionPickerColChangeEvent = await page.spyOnEvent('ionPickerColChange');
+
+      const column = page.locator('ion-picker-column');
+      const secondOption = column.locator('.picker-opt').nth(1);
+
+      await page.click('#single-column-button');
+
+      await ionPickerDidPresent.next();
+
+      secondOption.click();
+
+      await ionPickerColChangeEvent.next();
+
+      expect(ionPickerColChangeEvent.events[0].detail.prevSelected).toBe(0);
+      expect(ionPickerColChangeEvent.events[0].detail.selectedIndex).toBe(1);
```

Comment:
Decided to not use `toHaveReceivedEventDetail` because it returns the entire `col` object. The test only needs to test the `prevSelected` and `selectedIndex` value.